Formula/llvm.rb

```diff
@@ -298,7 +298,12 @@ def install
       system "cmake", "-G", "Unix Makefiles", "..", *(std_cmake_args + args)
       system "cmake", "--build", "."
       system "cmake", "--build", ".", "--target", "install"
-      system "cmake", "--build", ".", "--target", "install-xcode-toolchain" if MacOS::Xcode.installed?
+      if MacOS::Xcode.installed?
+        system "cmake", "--build", ".", "--target", "install-xcode-toolchain"
+        xctoolchain = prefix/"Toolchains/LLVM#{version}.xctoolchain"
```

Comment:
Hello. I am no developer. But I am stuck in this step installing llvm. 
Can you please tell me where should I find the duplicated files so I can delete them? 
Thanks.

`a==> Installing dependencies for node: llvm
==> Installing node dependency: llvm
==> cmake -G Unix Makefiles .. -DLLVM_ENABLE_PROJECTS=clang;clang-tools-extra;lld;lldb;mlir;polly -DLLVM_ENABLE_RUNTIMES=compiler-rt;libcxx;libcxxabi;libunw
==> cmake --build .
Last 15 lines from /Users/terryshultz/Library/Logs/Homebrew/llvm/02.cmake:
cd /tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/ELF && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ -DLLD_VENDOR=\"Homebrew\" -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/ELF -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/ELF -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/include -stdlib=libc++ -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-class-memaccess -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -O3 -DNDEBUG -std=c++14 -MD -MT tools/lld/ELF/CMakeFiles/lldELF.dir/Symbols.cpp.o -MF CMakeFiles/lldELF.dir/Symbols.cpp.o.d -o CMakeFiles/lldELF.dir/Symbols.cpp.o -c /tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/ELF/Symbols.cpp
[ 79%] Building CXX object tools/lld/ELF/CMakeFiles/lldELF.dir/SyntheticSections.cpp.o
cd /tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/ELF && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ -DLLD_VENDOR=\"Homebrew\" -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/ELF -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/ELF -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/include -stdlib=libc++ -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-class-memaccess -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -O3 -DNDEBUG -std=c++14 -MD -MT tools/lld/ELF/CMakeFiles/lldELF.dir/SyntheticSections.cpp.o -MF CMakeFiles/lldELF.dir/SyntheticSections.cpp.o.d -o CMakeFiles/lldELF.dir/SyntheticSections.cpp.o -c /tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/ELF/SyntheticSections.cpp
[ 79%] Building CXX object tools/lld/ELF/CMakeFiles/lldELF.dir/Target.cpp.o
cd /tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/ELF && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ -DLLD_VENDOR=\"Homebrew\" -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/ELF -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/ELF -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/include -stdlib=libc++ -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-class-memaccess -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -O3 -DNDEBUG -std=c++14 -MD -MT tools/lld/ELF/CMakeFiles/lldELF.dir/Target.cpp.o -MF CMakeFiles/lldELF.dir/Target.cpp.o.d -o CMakeFiles/lldELF.dir/Target.cpp.o -c /tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/ELF/Target.cpp
[ 79%] Building CXX object tools/lld/ELF/CMakeFiles/lldELF.dir/Thunks.cpp.o
cd /tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/ELF && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ -DLLD_VENDOR=\"Homebrew\" -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/ELF -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/ELF -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/include -stdlib=libc++ -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-class-memaccess -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -O3 -DNDEBUG -std=c++14 -MD -MT tools/lld/ELF/CMakeFiles/lldELF.dir/Thunks.cpp.o -MF CMakeFiles/lldELF.dir/Thunks.cpp.o.d -o CMakeFiles/lldELF.dir/Thunks.cpp.o -c /tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/ELF/Thunks.cpp
[ 79%] Building CXX object tools/lld/ELF/CMakeFiles/lldELF.dir/Writer.cpp.o
cd /tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/ELF && /usr/local/Homebrew/Library/Homebrew/shims/mac/super/clang++ -DLLD_VENDOR=\"Homebrew\" -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/ELF -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/ELF -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/include -I/tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/include -stdlib=libc++ -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-class-memaccess -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -O3 -DNDEBUG -std=c++14 -MD -MT tools/lld/ELF/CMakeFiles/lldELF.dir/Writer.cpp.o -MF CMakeFiles/lldELF.dir/Writer.cpp.o.d -o CMakeFiles/lldELF.dir/Writer.cpp.o -c /tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/lld/ELF/Writer.cpp
[ 79%] Linking CXX static library ../../../lib/liblldELF.a
cd /tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/ELF && /usr/local/Cellar/cmake/3.21.4/bin/cmake -P CMakeFiles/lldELF.dir/cmake_clean_target.cmake
cd /tmp/llvm-20211103-1783-139ffud/llvm-project-13.0.0.src/llvm/build/tools/lld/ELF && /usr/local/Cellar/cmake/3.21.4/bin/cmake -E cmake_link_script CMakeFiles/lldELF.dir/link.txt --verbose=1
"/Library/Developer/CommandLineTools/usr/bin/libtool" -static -no_warning_for_no_symbols -o ../../../lib/liblldELF.a CMakeFiles/lldELF.dir/AArch64ErrataFix.cpp.o CMakeFiles/lldELF.dir/Arch/AArch64.cpp.o CMakeFiles/lldELF.dir/Arch/AMDGPU.cpp.o CMakeFiles/lldELF.dir/Arch/ARM.cpp.o CMakeFiles/lldELF.dir/Arch/AVR.cpp.o CMakeFiles/lldELF.dir/Arch/Hexagon.cpp.o CMakeFiles/lldELF.dir/Arch/Mips.cpp.o CMakeFiles/lldELF.dir/Arch/MipsArchTree.cpp.o CMakeFiles/lldELF.dir/Arch/MSP430.cpp.o CMakeFiles/lldELF.dir/Arch/PPC.cpp.o CMakeFiles/lldELF.dir/Arch/PPC64.cpp.o CMakeFiles/lldELF.dir/Arch/RISCV.cpp.o CMakeFiles/lldELF.dir/Arch/SPARCV9.cpp.o CMakeFiles/lldELF.dir/Arch/X86.cpp.o CMakeFiles/lldELF.dir/Arch/X86_64.cpp.o CMakeFiles/lldELF.dir/ARMErrataFix.cpp.o CMakeFiles/lldELF.dir/CallGraphSort.cpp.o CMakeFiles/lldELF.dir/DWARF.cpp.o CMakeFiles/lldELF.dir/Driver.cpp.o CMakeFiles/lldELF.dir/DriverUtils.cpp.o CMakeFiles/lldELF.dir/EhFrame.cpp.o CMakeFiles/lldELF.dir/ICF.cpp.o CMakeFiles/lldELF.dir/InputFiles.cpp.o CMakeFiles/lldELF.dir/InputSection.cpp.o CMakeFiles/lldELF.dir/LTO.cpp.o CMakeFiles/lldELF.dir/LinkerScript.cpp.o CMakeFiles/lldELF.dir/MapFile.cpp.o CMakeFiles/lldELF.dir/MarkLive.cpp.o CMakeFiles/lldELF.dir/OutputSections.cpp.o CMakeFiles/lldELF.dir/Relocations.cpp.o CMakeFiles/lldELF.dir/ScriptLexer.cpp.o CMakeFiles/lldELF.dir/ScriptParser.cpp.o CMakeFiles/lldELF.dir/SymbolTable.cpp.o CMakeFiles/lldELF.dir/Symbols.cpp.o CMakeFiles/lldELF.dir/SyntheticSections.cpp.o CMakeFiles/lldELF.dir/Target.cpp.o CMakeFiles/lldELF.dir/Thunks.cpp.o CMakeFiles/lldELF.dir/Writer.cpp.o
[ 79%] Built target lldELF
make: *** [all] Error 2`

Reply:
That's an unrelated issue and very clearly tells you "Do not report this issue to Homebrew/brew or Homebrew/core!"

Reply:
Well thanks. It printed the link to the issue. Sorry about that.

Reply:
Well thanks. It printed the link to the issue. Sorry about that.

Reply:
If you just need `llvm` to build `node`, you can probably get away with getting rid of everything but `clang` in the `projects` array, like so:
```diff
diff --git a/Formula/llvm.rb b/Formula/llvm.rb
index b7912091ec3..83b5cbc59e0 100644
--- a/Formula/llvm.rb
+++ b/Formula/llvm.rb
@@ -55,11 +55,6 @@ class Llvm < Formula
   def install
     projects = %w[
       clang
-      clang-tools-extra
-      lld
-      lldb
-      mlir
-      polly
     ]
     runtimes = %w[
       compiler-rt
```

You'll want to do `git -C "$(brew --repository homebrew/core)" restore Formula/llvm.rb` when you're done to prevent `brew update` weirdness.

**Note:** I haven't tested this, so I don't know if it will work. However, it might help. If it doesn't work, however, then there's not much more I can do to help you. Sorry!

Reply:
Thank you